sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/aggregation/CountAggregation.java

```diff
@@ -10,19 +10,20 @@
 import io.opentelemetry.sdk.metrics.aggregator.AggregatorFactory;
 import io.opentelemetry.sdk.metrics.aggregator.CountAggregator;
 import io.opentelemetry.sdk.metrics.common.InstrumentDescriptor;
-import io.opentelemetry.sdk.metrics.common.InstrumentValueType;
 import io.opentelemetry.sdk.metrics.data.MetricData;
 import io.opentelemetry.sdk.resources.Resource;
 import java.util.List;
 import java.util.Map;
 import javax.annotation.concurrent.Immutable;
 
 @Immutable
-enum CountAggregation implements Aggregation {
-  INSTANCE;
+final class CountAggregation implements Aggregation {
+  static final CountAggregation INSTANCE = new CountAggregation();
+
+  private CountAggregation() {}
 
   @Override
-  public AggregatorFactory getAggregatorFactory(InstrumentValueType instrumentValueType) {
+  public AggregatorFactory getAggregatorFactory() {
```

Comment:
I wonder if we could hide the `AggregatorFactory` on the Aggregation and just have a method like `getAggregator()` on it directly, so simplify the call chain. I might be missing something, though.

Reply:
Will look later, I think I need the factory, but you will see that probably needs a different name.